sda-commons-server-testing/src/main/java/org/sdase/commons/server/testing/junit5/EnvironmentExtension.java

```diff
@@ -0,0 +1,75 @@
+package org.sdase.commons.server.testing.junit5;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.sdase.commons.server.testing.Environment;
+
+/**
+ * This Junit 5 extension allows to set environment variables for JUnit tests. All environment
+ * variables will be reset after the test has run.
+ *
+ * <p>Example:
+ *
+ * <pre>
+ *   class MyTest {
+ *
+ *     &#64;RegisterExtension
+ *     public final env = new EnvironmentExtension().setEnv("DISABLE_JWT", "true");
+ *
+ *   }
+ * </pre>
+ */
+public class EnvironmentExtension implements BeforeAllCallback, AfterAllCallback {
```

Comment:
Did you checked out if it's an option to use [system-rules](https://github.com/stefanbirkner/system-rules/issues/55), [system-lambda](https://github.com/stefanbirkner/system-lambda) or some kind of fork of it as @JoergSiebahn suggested?

Reply:
There is one project that we would like to use for the replacement of the RetryRule anyway: https://junit-pioneer.org/docs/
Unfortunately it does not offer the feature to set the value using  a supplier function (what we need to lazily set the values).

There is not a framework that I know of that offers exactly what we want here.

Reply:
I've [seen use of EnvironmentRule](https://github.com/search?p=1&q=org%3ASDA-SE+EnvironmentRule&type=Code) that needs lazy values in sda commons and some archived services like [otp-service|https://github.com/SDA-SE/otp-service/search?q=TWILIO_URL] and some services where it can easily replaced with Dropwizard config overrides. All other occurrences in [SDA SE|https://github.com/SDA-SE] should be replaceable with the [annotations](https://junit-pioneer.org/docs/environment-variables/) of the mentioned lib as they use static values.

Reply:
otp-service is archived but I think we need the lazy feature anyways.
Would be an option to contribute that behaviour to "junit-pioneer" for example.
But that's nothing I would do rightaway in our current work to add JUnit 5 functionality.

Reply:
If we really need the lazy feature, the extension is fine for me.

According the system lambdas: As there is no explicit Junit5 extension available yet, we need to implement our own wrapper anyways. I do not see much benefit in wrapping another dependency over reusing (a copy of) our existing code.